controller/metrics/client.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	goclient "github.com/argoproj/argo-rollouts/utils/go-client"
+)
+
+const (
+	clientsetMetricsNamespace = "controller_clientset"
+)
+
+type K8sRequestsCountProvider struct {
+	k8sRequestsCount *prometheus.CounterVec
+}
+
+func (f *K8sRequestsCountProvider) Register(registerer prometheus.Registerer) {
+	f.k8sRequestsCount = k8sRequestsCount
+	registerer.MustRegister(f.k8sRequestsCount)
+}
+
+var (
+	// Custom events metric
+	k8sRequestsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: clientsetMetricsNamespace,
+			Name:      "k8s_request_total",
+			Help:      "Number of kubernetes requests executed during application reconciliation.",
+		},
+		[]string{"kind", "namespace", "name", "verb", "statusCode"},
```

Comment:
I think it's uncommon to use camelCase for prometheus metrics or labels, suggest status_code.